responses.py

```diff
@@ -640,9 +640,15 @@ def _on_request(self, adapter, request, **kwargs):
                 return _real_send(adapter, request, **kwargs)
 
             error_msg = (
-                "Connection refused by Responses: {0} {1} doesn't "
-                "match Responses Mock".format(request.method, request.url)
+                "Connection refused by Responses - the call doesn't "
+                "match any registered mock.\n\n"
+                "Request: \n"
+                "- {0} {1}\n\n"
+                "Available matches:\n".format(request.method, request.url)
             )
+            for m in self._matches:
+                error_msg += "- {} {}\n".format(m.method, m.url)
```

Comment:
```suggestion
                error_msg += u"- {} {}\n".format(m.method, m.url)
```
Could hit a UnicodeError here in Python 2